text/0000-stop-deleting-cache-images.md

```diff
@@ -0,0 +1,74 @@
+# Meta
+[meta]: #meta
+- Name: Stop deleting cache images
+- Start Date: 2022-03-31
+- Author(s): jabrown85
+- Status: Draft <!-- Acceptable values: Draft, Approved, On Hold, Superseded -->
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: [buildpacks/lifecycle#803](https://github.com/buildpacks/lifecycle/issues/803)
+- Supersedes: (put "N/A" unless this replaces an existing RFC, then link to that RFC)
+
+# Summary
+[summary]: #summary
+
+Lifecycle will no longer try to delete previous cache images when publishing cache images.
+
+# Definitions
+[definitions]: #definitions
+
+* Cache Image: The cache stored between builds - stored in a registry.
+* ECR: Amazon's container registry product
+
+# Motivation
+[motivation]: #motivation
+
+- Why should we do this?
+  As discussed in [buildpacks/lifecycle#803](https://github.com/buildpacks/lifecycle/issues/803), some registries (ECR) do not support `DELETE`. For platforms that work exclusively with such registries, the warning output by lifecycle and the time taken to fail is unavoidable and lifecycle is wasting time trying to complete an operation that will never succeed.
+
+- What use cases does it support?
+  All platforms that use cache images against registries that do not support delete will no longer see warning messages.
+
+- What is the expected outcome?
+  Platforms will need to handle cleanup of their cache images on their own, if they desire.
+
+# What it is
+[what-it-is]: #what-it-is
+
+Lifecycle will no longer attempt to delete cache images during cache image export.
+
+# How it Works
+[how-it-works]: #how-it-works
+
+Lifecycle will no longer attempt to delete cache images during cache image export.
+
+# Migration
+[migration]: #migration
+
+Lifecycle will document this behavior change in Release Notes/Changelog along with the associated Platform API that enables the new behavior.
```

Comment:
I'm not able to find any reference to the delete operation as part of the current Platform API spec. It seems like this is an implementation feature and should not have any impact on the Platform API.

If that is true, I would expect new versions of the lifecycle to stop deleting images regardless of Platform API version they are operating in.